### DISCUSSION TITLE
How to make the TextInput become readonly (not disabled)

### DISCUSSION THREAD

**Discussion by jigujigu** [filamentphp/filament#6330] [Help] [answered]
I need to make the TextInput become readony (not disabled), To make user more easily to read. 
Disable option make the text blur.

I tried using extraAttribut(), **but not work.**
`->extraAttributes(['readonly' => true])`

![image](https://user-images.githubusercontent.com/6214964/234757072-82add285-6429-45ff-bde7-e67fe31770c6.png)

any other good suggestion ? 
thank you

**Comment by danharrin** [comment, marked as answer]
Isn't there a `readonly()` method? If not, try `->extraInputAttributes(['readonly' => true])`
